src/org/rascalmpl/values/RascalFunctionValueFactory.java

```diff
@@ -583,7 +584,14 @@ private ITree parseObject(String methodName, ISourceLocation location, char[] in
                 recoverer = new ToTokenRecoverer(uri, parserInstance, new StackNodeIdDispenser(parserInstance));
                 //debugListener = new DebugLogger(new PrintWriter(System.out, true));
             }
-            return (ITree) parserInstance.parse(methodName, uri, input, exec, new DefaultNodeFlattener<>(), new UPTRNodeFactory(allowAmbiguity), recoverer, debugListener);
+            ITree parseForest = (ITree) parserInstance.parse(methodName, uri, input, exec, new DefaultNodeFlattener<>(), new UPTRNodeFactory(allowRecovery || allowAmbiguity), recoverer, debugListener);
+
+            if (!allowAmbiguity && allowRecovery && (filters == null || filters.isEmpty())) {
+                // Filter error-induced ambiguities
+                parseForest = (ITree) new ParseErrorDisambiguator((RascalValueFactory) ValueFactoryFactory.getValueFactory()).disambiguateErrors(parseForest, null);
```

Comment:
do we have to have  `null`? can't it be `VF.boolean(false)`? less nulls make the code a happier place.

Reply:
Yes we can. This was laziness on my part, I did not take the time to find out how to create an IBool.